ts/twitter/src/app/login/_components/use-password-modal.ts

```diff
@@ -0,0 +1,43 @@
+import { login } from "@/lib/actions/login";
+import { useRouter } from "next/navigation";
+import type { LoginFormValue } from "./login-form-types";
+
+interface UsePasswordModalReturn {
+  handleLoginAction: (
+    prevState: string | undefined,
+    formData: FormData,
+  ) => Promise<string | undefined>;
+}
+
+export const usePasswordModal = (
+  loginFormValue: LoginFormValue,
+): UsePasswordModalReturn => {
+  const router = useRouter();
+
+  const handleLoginAction = async (
+    prevState: string | undefined,
+    formData: FormData,
+  ) => {
+    try {
+      const username = formData.get("username") as string;
+      const password = formData.get("password") as string;
+
+      const result = await login({ username, password });
+
+      if (result.ok) {
+        router.push("/home");
+        return undefined;
+      }
+      return `Login failed: ${result.error.statusText}`;
```

Comment:
```suggestion
      if (!result.ok) {
        return `Login failed: ${result.error.statusText}`;
      }
      
      router.push("/home");
```
You don’t need to explicitly return undefined; you can write it like this instead.